### PR TITLE
test: Add BlockNodeAPITest w no mocks

### LINKED ISSUE #1773: Add no mock E2E Test class
### Story Form

As a block node engineer
I want to be able to run a real instance of the BN in an E2E without mocks
So that I can easily step through and replicate scenarios that involve multiple plugins

### Technical Notes

To date the team has noted it's been a bit challenging to test in an E2E fashion without first dockerizing due to the multi plugin logic.

It should be possible to spin up the BN with real instances and configs and walk E2E without mocks.

In the example test class add example of real API calls that initial the event driven architecture of the BN

### PR DESCRIPTION
**Pull request by Nana-EC**
## Reviewer Notes
There's currently no easy way to run the Block Node App in an E2E manner to easily step through the servers logic.
This will and does make debugging challenging and requires engineers utilize docker setups.
We want to have multiple options and don't want to be confined in our ability to test and triage.

- Add `BlockNodeAPITest` class that support E2E runs of BlockNodeApp w no mocks
- Add initial example test cases should how to write simple and complex tests to hit the server. Ensure logic initiates BN before tests and tears it down after
- Add `http2ClientPublishBlockStreamGet()` to should a call with a Helidon `Http2Client`
- Add `serverStatusRequestOnEmptyBN()` as examples of simple tests case using pbj built client to show an empty BN state
- Add `getBlockRequestNotAvailable()` as examples of simple tests case using pbj built client to show a `NOT_AVAILABLE` status on a block not present in a BN
- Add `e2eBlockStreamsBaseScenarios()` to show case multiple base scenarios that should always be functional

## Additional Notes ##
- Example CI run https://github.com/hiero-ledger/hiero-block-node/actions/runs/18851556933/job/53789166067?pr=1781
- `e2eBlockStreamsBaseScenarios()` is provided as an example to build more complex tests. A good test case to come out of this will all an an engineer to feed in real blocks to easily step through block processing. 

## Related Issue(s)
Fixes #1773

### CODE REVIEW

File: tools-and-tests/suites/src/main/resources/app-test.properties [hunk 1]
```diff
@@ -0,0 +1,7 @@
+# SPDX-License-Identifier: Apache-2.0
+
+prometheus.endpointEnabled=false
+mediator.ringBufferSize=1024
+notifier.ringBufferSize=1024
```
Comment (AlfredoG87) [inline tools-and-tests/suites/src/main/resources/app-test.properties]: these properties are no longer used anywere.

**Review by jsync-swirlds — DISMISSED** [review]

**Review by jsync-swirlds — APPROVED** [review]

File: tools-and-tests/suites/src/main/java/org/hiero/block/suites/e2e/BlockNodeAPITests.java [hunk 2]
```diff
@@ -0,0 +1,426 @@
...
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.time.Duration;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.concurrent.CountDownLatch;
+import java.util.function.Function;
+import org.hiero.block.api.*;
```
Comment (jsync-swirlds) [inline tools-and-tests/suites/src/main/java/org/hiero/block/suites/e2e/BlockNodeAPITests.java]: Should not be using wildcard imports.
Most IDEs have a setting to collapse these after some (small by default) number of imports; that may need to be adjusted.
Example (from IntelliJ, note 999 used to prevent wildcards)
image

File: tools-and-tests/suites/src/main/java/org/hiero/block/suites/e2e/BlockNodeAPITests.java [hunk 3]
```diff
@@ -0,0 +1,426 @@
...
+ * E2E test class that performs a real-world instantiation of BlockNodeApp with no mocks.
+ * It uses real services loaded via ServiceLoaderFunction and a real Helidon webserver.
+ * Tests be used to confirm basic gRPC operations and follow complex multi plugin flows against the running server.
+ * Both Helidon Http2Client and PbjGrpcClient approaches are provided to perform gRPC requests.
+ * Cleans up the BlockNodeApp and the store of blocks after each test.
+ */
+@Tag("api")
+public class BlockNodeAPITests {
+
+    private static String BLOCKS_DATA_DIR_PATH = "build/tmp/data";
```
Comment (AlfredoG87) [inline tools-and-tests/suites/src/main/java/org/hiero/block/suites/e2e/BlockNodeAPITests.java]: what is this directory used for?

I see we deleted everything in it before each test, but I don't see we use it anywhere... what is the intention?

In some other tests we use this:

```
    @TempDir
    private Path tempDir;

```

have you considered it as an alternative?


depending on the intention of the folder I would favor one or the other.

File: tools-and-tests/suites/src/main/java/org/hiero/block/suites/utils/BlockItemBuilderUtils.java [hunk 4]
```diff
@@ -0,0 +1,372 @@
...
+    }
+
+    /**
+     * Creates a sample BlockItem representing a block header with the given block number and consensus time.
+     */
+    public static BlockItem sampleBlockHeader(final long blockNumber, Instant consensusTime) {
+        return new BlockItem(new OneOf<>(
+                ItemOneOfType.BLOCK_HEADER,
+                new BlockHeader(
+                        new SemanticVersion(1, 2, 3, "a", "b"),
```
Comment (AlfredoG87) [inline tools-and-tests/suites/src/main/java/org/hiero/block/suites/utils/BlockItemBuilderUtils.java]: same here. these hapi version must be intentional to the test case

File: tools-and-tests/suites/src/main/java/org/hiero/block/suites/utils/BlockItemBuilderUtils.java [hunk 5]
```diff
@@ -0,0 +1,372 @@
...
+     * Creates an array of BlockItem objects representing a very simple block stream of blocks from startBlockNumber to
+     * but not including endBlockNumber.
+     *
+     * @param startBlockNumber the starting block number
+     * @param endBlockNumber the ending block number, inclusive
+     * @param firstBlockConsensusTime the consensus time of the first block
+     * @param consensusTimeBetweenBlocks the time between blocks starts with the first block at 2025-01-01T00:00:00Z
+     * @return an array of BlockItem objects
+     */
+    public static BlockItemUnparsed[] createNumberOfVerySimpleBlocksUnparsed(
```
Comment (AlfredoG87) [inline tools-and-tests/suites/src/main/java/org/hiero/block/suites/utils/BlockItemBuilderUtils.java]: This method seems unused, are you planning to use it in the future?

File: tools-and-tests/suites/src/main/java/org/hiero/block/suites/utils/BlockItemBuilderUtils.java [hunk 6]
```diff
@@ -0,0 +1,372 @@
...
+        for (int i = 0; i < blockItems.length; i += 3) {
+            long blockNumber = i / 3;
+            blockItems[i] = sampleBrokenBlockHeaderUnparsed(blockNumber);
+            blockItems[i + 1] = sampleRoundHeaderUnparsed(blockNumber * 10);
+            blockItems[i + 2] = sampleBlockProofUnparsed(blockNumber);
+        }
+        return blockItems;
+    }
+
+    public static BlockItemUnparsed[] createNumberOfVerySimpleBlocksUnparsedWithBrokenProofs(
```
Comment (AlfredoG87) [inline tools-and-tests/suites/src/main/java/org/hiero/block/suites/utils/BlockItemBuilderUtils.java]: This method seems unused, are you planning to use it in the future?

File: tools-and-tests/suites/src/main/java/org/hiero/block/suites/utils/BlockItemBuilderUtils.java [hunk 7]
```diff
@@ -0,0 +1,372 @@
...
+        for (int i = 0; i < blockItems.length; i += 3) {
+            long blockNumber = i / 3;
+            blockItems[i] = sampleBlockHeaderUnparsed(blockNumber);
+            blockItems[i + 1] = sampleRoundHeaderUnparsed(blockNumber * 10);
+            blockItems[i + 2] = sampleBrokenBlockProofUnparsed(blockNumber);
+        }
+        return blockItems;
+    }
+
+    public static BlockItemUnparsed[] createNumberOfVerySimpleBlocksUnparsedWithNullHeaderBytes(
```
Comment (AlfredoG87) [inline tools-and-tests/suites/src/main/java/org/hiero/block/suites/utils/BlockItemBuilderUtils.java]: This method seems unused, are you planning to use it in the future?

File: tools-and-tests/suites/src/main/java/org/hiero/block/suites/utils/BlockItemBuilderUtils.java [hunk 8]
```diff
@@ -0,0 +1,372 @@
...
+        for (int i = 0; i < blockItems.length; i += 3) {
+            long blockNumber = i / 3;
+            blockItems[i] = sampleNullBlockHeaderUnparsed();
+            blockItems[i + 1] = sampleRoundHeaderUnparsed(blockNumber * 10);
+            blockItems[i + 2] = sampleBlockProofUnparsed(blockNumber);
+        }
+        return blockItems;
+    }
+
+    public static BlockItems[] createNumberOfSimpleBlockBatches(final int numberOfBatches) {
```
Comment (AlfredoG87) [inline tools-and-tests/suites/src/main/java/org/hiero/block/suites/utils/BlockItemBuilderUtils.java]: This method seems unused, are you planning to use it in the future?

File: tools-and-tests/suites/src/main/java/org/hiero/block/suites/utils/BlockItemBuilderUtils.java [hunk 9]
```diff
@@ -0,0 +1,372 @@
...
+
+    public static BlockItem[] createSimpleBlockWithNumber(final long blockNumber) {
+        final BlockItem[] blockItems = new BlockItem[3];
+        blockItems[0] = sampleBlockHeader(blockNumber);
+        blockItems[1] = sampleRoundHeader(blockNumber * 10L);
+        blockItems[2] = sampleBlockProof(blockNumber);
+        return blockItems;
+    }
+
+    public static BlockItemUnparsed[] createSimpleBlockUnparsedWithNumber(final long blockNumber) {
```
Comment (AlfredoG87) [inline tools-and-tests/suites/src/main/java/org/hiero/block/suites/utils/BlockItemBuilderUtils.java]: This method seems unused, are you planning to use it in the future?

File: tools-and-tests/suites/src/main/java/org/hiero/block/suites/utils/BlockItemBuilderUtils.java [hunk 10]
```diff
@@ -0,0 +1,372 @@
...
+import com.hedera.pbj.runtime.io.buffer.Bytes;
+import java.time.Duration;
+import java.time.Instant;
+import java.util.*;
+import org.hiero.block.internal.BlockItemUnparsed;
+import org.hiero.block.node.spi.blockmessaging.BlockItems;
+
+/**
+ * A utility class to create sample BlockItem objects for testing purposes.
+ * Inspired by BlockItemBuilderUtils in block-node-app testFixtures.
```
Comment (AlfredoG87) [inline tools-and-tests/suites/src/main/java/org/hiero/block/suites/utils/BlockItemBuilderUtils.java]: are we not repeating too much similar test utils?
could we have a single place for creating test blocks?

File: tools-and-tests/suites/src/main/java/org/hiero/block/suites/e2e/BlockNodeAPITests.java [hunk 11]
```diff
@@ -0,0 +1,426 @@
...
+    private final Function<PublishStreamResponse, Long> endStreamBlockNumberExtractor =
+            response -> Objects.requireNonNull(response.endStream()).blockNumber();
+
+    private record Options(Optional<String> authority, String contentType) implements ServiceInterface.RequestOptions {}
+
+    private BlockNodeApp app;
+    private PbjGrpcClient publishBlockStreamPbjGrpcClient;
+    private PbjGrpcClient subscribeBlockStreamPbjGrpcClient;
+    private PbjGrpcClient serverStatusPbjGrpcClient;
+    private PbjGrpcClient getBlockPbjGrpcClient;
```
Comment (AlfredoG87) [inline tools-and-tests/suites/src/main/java/org/hiero/block/suites/e2e/BlockNodeAPITests.java]: is it necessary to create a grpcClient per API?

**Review by jsync-swirlds — APPROVED** [review]

File: tools-and-tests/suites/src/main/java/org/hiero/block/suites/e2e/BlockNodeAPITests.java [hunk 12]
```diff
@@ -0,0 +1,439 @@
...
+            endStreamResponseCodeExtractor =
+                    response -> Objects.requireNonNull(response.endStream()).status();
+    private final Function<PublishStreamResponse, Long> endStreamBlockNumberExtractor =
+            response -> Objects.requireNonNull(response.endStream()).blockNumber();
+
+    private record Options(Optional<String> authority, String contentType) implements ServiceInterface.RequestOptions {}
+
+    private BlockNodeApp app;
+    private PbjGrpcClient publishBlockStreamPbjGrpcClient;
+    private PbjGrpcClient subscribeBlockStreamPbjGrpcClient;
```
Comment (AlfredoG87) [inline tools-and-tests/suites/src/main/java/org/hiero/block/suites/e2e/BlockNodeAPITests.java]: would not make more sense to have the 

private BlockStreamPublishServiceInterface.BlockStreamPublishServiceClient instead of the generic PbjGrpcClient?

File: tools-and-tests/suites/src/main/java/org/hiero/block/suites/e2e/BlockNodeAPITests.java [hunk 13]
```diff
@@ -0,0 +1,439 @@
...
+        assertThat(initialResponseObserver.getOnNextCalls()).hasSize(0); // no responses should be sent yet
+
+        // Assert no other responses sent
+        assertThat(initialResponseObserver.getOnErrorCalls()).isEmpty();
+        assertThat(initialResponseObserver.getOnSubscriptionCalls()).isEmpty();
+        assertThat(initialResponseObserver.getOnCompleteCalls().get()).isEqualTo(0);
+        assertThat(initialResponseObserver.getClientEndStreamCalls().get()).isEqualTo(0);
+
+        // close the initial connection
+        publishBlockStreamPbjGrpcClient.close(); // to-do: figure out why this close is not triggering a block close
```
Comment (AlfredoG87) [inline tools-and-tests/suites/src/main/java/org/hiero/block/suites/e2e/BlockNodeAPITests.java]: after rebase, if you enable this test and add this:


```suggestion
        // send endStream before closing connection
        PublishStreamRequest endStreamRequest = PublishStreamRequest.newBuilder()
                .endStream(PublishStreamRequest.EndStream.newBuilder().endCode(PublishStreamRequest.EndStream.Code.RESET)
                        .build())
                .build();
        requestStream.onNext(endStreamRequest);

        publishBlockStreamPbjGrpcClient.close(); // to-do: figure out why this close is not triggering a block close
```

and the next comment:

**Review by AlfredoG87 — APPROVED** [review]
First pass: I love the ability to quickly test e2e scenarios, leaved some questions, suggestions and comments. 

I was able to rebase, debug and troubleshoot the test `swappedPublisherConnection` was nice and fast. 

I am not sure if this is ready to be merged yet, but we can do incremental improvements, and maybe start replacing some e2e tests that use the old approach to this approach.

File: .editorconfig [hunk 14]
```diff
@@ -5,7 +5,7 @@ root = true
 
 [*]
 indent_style = space
-indent_size = 2
+indent_size = 4
```
[the remark this answers is not in the document]
Comment (jsync-swirlds) [reply .editorconfig]: 2 space indent doesn't match our style guide, this just makes users of editorconfig consistent.

[the remark this answers is not in the document]
Comment (AlfredoG87) [reply .editorconfig]: 👍

File: tools-and-tests/suites/src/main/java/org/hiero/block/suites/utils/BlockItemBuilderUtils.java [hunk 15]
```diff
@@ -0,0 +1,372 @@
...
+        return BlockItemUnparsed.newBuilder().blockProof(Bytes.wrap(arr)).build();
+    }
+
+    /**
+     * Creates an array of BlockItem objects representing a very simple block stream of N blocks.
+     *
+     * @param numberOfBlocks the number of blocks to create
+     * @return an array of BlockItem objects
+     */
+    public static BlockItem[] createNumberOfVerySimpleBlocks(final int numberOfBlocks) {
```
[the remark this answers is not in the document]
Comment (Nana-EC) [reply tools-and-tests/suites/src/main/java/org/hiero/block/suites/utils/BlockItemBuilderUtils.java]: Left over from class copy. I imagined with more tests we'd also end up needing it so left it

File: tools-and-tests/suites/src/main/java/org/hiero/block/suites/e2e/BlockNodeAPITests.java (hunk 11 above)
Comment (Nana-EC) [reply tools-and-tests/suites/src/main/java/org/hiero/block/suites/e2e/BlockNodeAPITests.java]: I wanted to avoid crossover and allow for test developers to customize their client without impacting the whole class. Was planning to add parameters to the method

File: tools-and-tests/suites/src/main/java/org/hiero/block/suites/e2e/BlockNodeAPITests.java [hunk 16]
```diff
@@ -0,0 +1,426 @@
...
+        if (Files.exists(dataDir)) {
+            // delete all files and subdirectories
+            Files.walk(dataDir)
+                    .sorted(Comparator.reverseOrder())
+                    .map(Path::toFile)
+                    .forEach(java.io.File::delete);
+        }
+
+        // Use the real service loader so all real implementations are loaded
+        app = new BlockNodeApp(new ServiceLoaderFunction(), false);
```
[the remark this answers is not in the document]
Comment (Nana-EC) [reply tools-and-tests/suites/src/main/java/org/hiero/block/suites/e2e/BlockNodeAPITests.java]: Good questions. Thinking on the custom config considerations. I'm hoping it doesn't come down to env variable overrides.
For Pikachu I need to think more

[the remark this answers is not in the document]
Comment (AlfredoG87) [reply tools-and-tests/suites/src/main/java/org/hiero/block/suites/e2e/BlockNodeAPITests.java]: Yeah, we can all think about it and do incremental improvements here.

File: tools-and-tests/suites/src/main/java/org/hiero/block/suites/e2e/BlockNodeAPITests.java (hunk 13 above)
Comment (jsync-swirlds) [reply tools-and-tests/suites/src/main/java/org/hiero/block/suites/e2e/BlockNodeAPITests.java]: Note, this requires the work _planned_ under Issue #1239, it won't work _yet_.

File: tools-and-tests/suites/src/main/java/org/hiero/block/suites/e2e/BlockNodeAPITests.java (hunk 3 above)
Comment (jsync-swirlds) [reply tools-and-tests/suites/src/main/java/org/hiero/block/suites/e2e/BlockNodeAPITests.java]: We typically try to use `@TempDir` because it's safer. It's way too easy to accidentally end up with files from a temp dir in the codebase added to git; and not all systems reliably handle those files; the annotation is environment aware and more reliable.
